recipes-extended/xen/files/libxl-stubdom-options.patch

```diff
@@ -0,0 +1,119 @@
+CHANGELOG
+################################################################################
+Authors:
+Chris Rogers <rogersc@ainfosec.com>
+
+################################################################################
+REMOVAL
+################################################################################
+This patch should be removed if the stubdom work is upstreamed.
+
+################################################################################
+UPSTREAM PLAN
+################################################################################
+Should be upstreamed with the rest of stubdom work when appropriate.
+
+################################################################################
+INTERNAL DEPENDENCIES
+################################################################################
+
+################################################################################
+PATCHES
+################################################################################
+
+Index: xen-4.6.1/tools/libxl/libxl_types.idl
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_types.idl
++++ xen-4.6.1/tools/libxl/libxl_types.idl
+@@ -462,6 +462,11 @@ libxl_domain_build_info = Struct("domain
+     #  parameters for all type of scheduler
+     ("sched_params",     libxl_domain_sched_params),
+ 
++    #Memory size for stubdom
++    ("stubdom_memory",  MemKB),
++    #Kernel args for stubdom
++    ("stubdom_cmdline", string),
++
+     ("ioports",          Array(libxl_ioport_range, "num_ioports")),
+     ("irqs",             Array(uint32, "num_irqs")),
+     ("iomem",            Array(libxl_iomem_range, "num_iomem")),
+Index: xen-4.6.1/tools/libxl/libxl_dm.c
+===================================================================
+--- xen-4.6.1.orig/tools/libxl/libxl_dm.c
++++ xen-4.6.1/tools/libxl/libxl_dm.c
+@@ -1433,6 +1433,7 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+         abort();
+     }
+     dm_config->b_info.max_memkb += guest_config->b_info.video_memkb;
++    dm_config->b_info.max_memkb += guest_config->b_info.stubdom_memory;
+     dm_config->b_info.target_memkb = dm_config->b_info.max_memkb;
+ 
+     dm_config->b_info.u.pv.features = "";
+@@ -1444,6 +1445,7 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+     dm_config->b_info.extra = guest_config->b_info.extra;
+     dm_config->b_info.extra_pv = guest_config->b_info.extra_pv;
+     dm_config->b_info.extra_hvm = guest_config->b_info.extra_hvm;
++    dm_config->b_info.stubdom_cmdline = guest_config->b_info.stubdom_cmdline;
+ 
+     dm_config->disks = guest_config->disks;
+     dm_config->num_disks = guest_config->num_disks;
+@@ -1499,6 +1501,7 @@ void libxl__spawn_stub_dm(libxl__egc *eg
+      */
+     if (guest_config->b_info.stubdomain_version == LIBXL_STUBDOMAIN_VERSION_LINUX) {
+         int dmargs_size = 0;
++        char delim[] = "obvious-delimiter-pattern";
```

Comment:
Since this is a delimiter on kernel commandline and searched for by shell script, you could just use two bytes 0x03 0x04 <ETX><EOT>. If printable chars are still preferred, you could just use ETX, EOT or ETXEOT which uses a lot less chars, highly unlikely to occur in valid parameters, most people will know else quick google will explain.